en/lessons/visualizing-with-bokeh.md

```diff
@@ -0,0 +1,779 @@
+
+The intended uses of matplotlib and Bokeh are quite different. Matplotlib creates static graphics that are useful for quick and simple visualizations, or for creating publication quality images. Bokeh creates visualizations for display on the web (whether locally or embedded in a webpage) and most importantly, the visualizations are meant to be highly interactive. Matplotlib does not offer either of these features. 
+
+If would you like to visually interact with your data in an exploratory manner or you would like to distribute interactive visual data to a web audience, Bokeh is the library for you! If your main interest is producing finalized visualizations for publication, matplotlib may be better, although Bokeh does offer a way to create static graphics.
+
+With this differences in mind, as we work through the lesson, I'll emphasize the interactive aspects that make Bokeh useful for exploring and disseminating historical data and that set it apart from other libraries like matplotlib.
+
+## Your First Plot
+
+First, create a new file called `my_first_plot.py` in the same directory as  `wwii_thor.csv` and then open it up in a text editor. We'll be adding lines to this file to run. 
+
+```python
+#my_first_plot.py
+from bokeh.plotting import figure, output_file, show	
+```
+
+To implement and use Bokeh, we first import some basics that we need from the `bokeh.plotting` module.
+
+`figure` is the core object that we will use to create plots. `figure` handles the styling of plots, including title, labels, axes, and grids, and it exposes methods for adding data to the plot. The `output_file` function defines how the visualization will be rendered (namely to an html file) and the `show` function will be invoked when the plot is ready for output. `show` tells Bokeh that all of the data has been added to the plot and it is time to render it.
+
+```python
+output_file('my_first_graph.html')
+```
+
+Bokeh recommends that `output_file`, to which we pass a file name, be called at the start of your script, immediately after imports. An alternative output function to be aware of is `output_notebook` which is used to show plots in-line in a Jupyter Notebook. To learn more about installing and using Jupyter notebooks, see [Jupyter's documentation](https://jupyter.readthedocs.io/en/latest/).
+
+```python
+x = [1, 3, 5, 7]
+y = [2, 4, 6, 8]
+```
+
+Next we'll create some data to plot. Data in Bokeh can take on different forms, but at its simplest, data is just a list of values. We create one list for our x-axis and one for our y-axis.
+
+With our output format and data fixed, we can instantiate a `figure` and  add the data to it.
+
+```python
+p = figure()
+
+p.circle(x, y, size=10, color='red', legend='circle')
+p.line(x, y, color='blue', legend='line')
+p.triangle(y, x, color='yellow', size=10, legend='triangle')
+```
+
+<div class="alert alert-warning">
+
+`p` is a common variable name for a `figure` object, since a figure is a type of plot.
```

Comment:
Unfortunately, you cannot use markdown _within_ HTML tags in a markdown document like this. I'll create an include that'll let you do this, however.